### DISCUSSION TITLE
Viewing Job Outputs in the UI

### DISCUSSION THREAD

**Discussion by ashrafgt** [community/community#10430] [Actions]
Currently, it's possible to expose a set of output values from a certain job using syntax like:
```bash
echo "::set-output name=foo::$BAR"
```

Unfortunately, it's not possible (at least as far as I was able to see) to review this value from the Actions UI. I think it would be quite useful to show this value in order to help with debugging and understanding workflows. Right now, the workaround is, of course, trivial:
```bash
echo "::set-output name=foo::$BAR"
echo "$BAR"  # add this line
```
But it would be lovely to have this done by default unless the output is marked as secret.

**Comment by chrispat** [comment]
Hi @ashrafgt thank you for your feedback.

I agree it would be very useful to have the ability to see not only outputs from a given job but also inputs for a `workflow_dispatch` easily in the logs.  We have had this feature on our backlog for some time, however, we have not been able to prioritize it.  I will link this issue to our internal backlog so we can track the feedback.

**Comment by DavidIAm** [comment]
Concur.  I want to be able to see job outputs in the UI

**Comment by davidcedres** [comment]
In the next image, taken from https://github.blog/changelog/2020-12-08-github-actions-workflow-visualization/ I can see a link embedded in the UI (the job's output?), yet I cannot find how to replicate that.

![image](https://github.com/community/community/assets/7198245/343f7bbe-aaad-4a33-8943-a3a033b00773)

I wonder if somebody here can shed some light on this.

**Reply by humzam** [reply]
Hi @davidcedresx. That link embedded in the UI is a result of using GitHub Environments. When you configure a job to deploy to an environment, you can also specify a URL to show to the user, which is what you're seeing there.

**Comment by humzam** [comment]
> But it would be lovely to have this done by default unless the output is marked as secret.

@ashrafgt Just thought I'd add a tip. If you don't want to needlessly add all those `echo` commands in your workflow to see outputs, you can also just re-run your job and check the box labeled `Enable debug logging`. That will allow you to see all outputs and what value they receive. 

![Screenshot 2023-11-01 at 2 48 45 PM](https://github.com/community/community/assets/7700841/6ab978ab-ff5d-40bc-a205-a83f33a6d42f)

**Comment by simaotwx** [comment]
Still looking for a way to do this. Has anything changed since 2023?

EDIT:
There is something similar here: https://github.blog/news-insights/product-news/supercharging-github-actions-with-job-summaries/

**Comment by standaloneSA** [comment]
I am also finding that this would be helpful in introspection and debugging
